### PR TITLE
Club: Templates: Search, Category, Tag

### PR DESCRIPTION
**Pull request by matiasbenedetto**
#### Changes proposed in this Pull Request:
Implement archives template

#### Related issue(s):
https://github.com/Automattic/themes/issues/6221

### CODE REVIEW

File: club/patterns/post-list.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
```
Comment (scruffian) [inline club/patterns/post-list.php]: Is there a reason this can't be an HTML file?

**Review by madhusudhand — COMMENTED** [review]
It looks great. 
Following are few observations (I kept typography out of context because it would be taken care in different issues).

image

1. When post title is large, it breaks "read more" and date to next line. it should break the post title to next line keeping the date and button aligned with other posts.
2. post is not vertically center aligned. Also, it has different spacing bottom with and without featured image.
3. Uppercase the date.
4. Design suggests default button text as "READ IT"

File: club/patterns/post-list.php (hunk 1 above)
Comment (matiasbenedetto) [reply club/patterns/post-list.php]: I think we need PHP to make the strings translatable.

**Comment by matiasbenedetto** [comment]
The latest commits should cover these items:
> post is not vertically center aligned. Also, it has different spacing bottom with and without featured image.
> Uppercase the date.
> Design suggests default button text as "READ IT"

About the last remaining item:

> When post title is large, it breaks "read more" and date to next line. it should break the post title to next line keeping the date and button aligned with other posts.

I'm not sure if this is the expected behavior by watching the current design. Also it dosn't seem simple to achieve with CSS that would break the functionality of the editor.

I propose to approve this PR in its current state and try to follow up the last item as a refinement once the theme is more complete.

**Comment by madhusudhand** [comment]
> I'm not sure if this is the expected behavior by watching the current design. Also it dosn't seem simple to achieve with CSS that would break the functionality of the editor.

Archive page design wraps the post title without bringing the other items to next line.
image

I guess may not be possible to avoid CSS to achieve the design. I tried the following to achieve the designs with minimal css. 

1. Setting the flex wrap to nowrap for all `three row blocks`.

image

This brings them to same line, but it still breaks the text of buttons and buttons to multiple lines. Also image needs fixed width shrinks.

```
// image block
flex-shrink: 0;

// post-date and read-more blocks
white-space: nowrap;
```

image

It seems to be working. But needs more testing in other areas of the theme as well.
We could reserve these changes to another PR and test it well before merging.

**Review by madhusudhand — APPROVED** [review]
Everything else LGTM!

**Comment by matiasbenedetto** [comment]
Thanks for the testing @madhusudhand 
> Setting the flex wrap to nowrap for all three row blocks.


I think the problem with that approach is that it will work on large viewports but on mobile, it looks bad and it doesn't match with the wrap behavior present in the mockups. Example:

image

**Pull request merged**